data/data_loader.py

```diff
@@ -175,161 +151,37 @@ def __sort_and_precalc_image_ratings(self) -> tuple[float, list[float]]:
-                    if width * height < target_wh[0] * target_wh[1]:
-                        undersized_images.append(f" {pathname}, size: {width},{height}, target size: {target_wh}")
-
-                image_train_item = ImageTrainItem(image=None, # image loaded at runtime to apply jitter
-                                                  caption=caption, 
-                                                  target_wh=target_wh, 
-                                                  pathname=pathname, 
-                                                  flip_p=flip_p,
-                                                  multiplier=multipliers[current_dir],
-                                                 )
-
-                cur_file_multiplier = multipliers[current_dir]
-
-                while cur_file_multiplier >= 1.0:
-                    decorated_image_train_items.append(image_train_item)
-                    cur_file_multiplier -= 1
-                
-                if cur_file_multiplier > 0:
-                    if randomizer.random() < cur_file_multiplier:
-                        decorated_image_train_items.append(image_train_item)
-                
-            except Exception as e:
-                logging.error(f"{Fore.LIGHTRED_EX} *** Error opening {Fore.LIGHTYELLOW_EX}{pathname}{Fore.LIGHTRED_EX} to get metadata. File may be corrupt and will be skipped.{Style.RESET_ALL}")
-                logging.error(f" *** exception: {e}")
-                pass
 
         if not self.has_scanned:
             self.has_scanned = True
-            if len(undersized_images) > 0:
-                underized_log_path = os.path.join(self.log_folder, "undersized_images.txt")
-                logging.warning(f"{Fore.LIGHTRED_EX} ** Some images are smaller than the target size, consider using larger images{Style.RESET_ALL}")
-                logging.warning(f"{Fore.LIGHTRED_EX} ** Check {underized_log_path} for more information.{Style.RESET_ALL}")
-                with open(underized_log_path, "w") as undersized_images_file:
-                    undersized_images_file.write(f" The following images are smaller than the target size, consider removing or sourcing a larger copy:")
-                    for undersized_image in undersized_images:
-                        undersized_images_file.write(f"{undersized_image}\n")
-        
-        print (f" * DLMA: {len(decorated_image_train_items)} images loaded from {len(image_paths)} files")
-
-        return decorated_image_train_items
+            
+            logging.info(" Preloading images...")
+            
+            items, events = resolver.resolve(self.data_root, self.aspects, flip_p=flip_p, seed=self.seed)
+            image_paths = set(map(lambda item: item.pathname, items))
+
+            print (f" * DLMA: {len(items)} images loaded from {len(image_paths)} files")
+            
+            self.prepared_train_data = items
+            random.Random(self.seed).shuffle(self.prepared_train_data)
```

Comment:
Since `self.prepared_train_data` is what is used by the instance, I got rid of `self.image_paths` as a property since it is only used in this location.